### DISCUSSION TITLE
Need help for upgrading Wagtail 5.2.3 to 6.0

### DISCUSSION THREAD

**Discussion by arshyia3000** [wagtail/wagtail#11612] [Q&A] [answered]
Hey guys,

when trying to merge, I get dozen lines of these errors, upgrading from 5.2.3 to 6.0.

There should be some problems with my git merge. I haven't changed any file of wagtail codes in my local development.
I think everything is good from wagtail side. any idea how to merge properly? 

warning: Cannot merge binary files: wagtail/snippets/locale/pl/LC_MESSAGES/django.mo (HEAD vs. stable/6.0.x)
warning: Cannot merge binary files: wagtail/snippets/locale/nl/LC_MESSAGES/django.mo (HEAD vs. stable/6.0.x)
warning: Cannot merge binary files: wagtail/snippets/locale/nb/LC_MESSAGES/django.mo (HEAD vs. stable/6.0.x)
warning: Cannot merge binary files: wagtail/snippets/locale/mn/LC_MESSAGES/django.mo (HEAD vs. stable/6.0.x)
warning: Cannot merge binary files: wagtail/snippets/locale/mi/LC_MESSAGES/django.mo (HEAD vs. stable/6.0.x)
warning: Cannot merge binary files: wagtail/snippets/locale/lv/LC_MESSAGES/django.mo (HEAD vs. stable/6.0.x)

CONFLICT (content): Merge conflict in wagtail/snippets/locale/sr/LC_MESSAGES/django.po
Auto-merging wagtail/snippets/locale/sv/LC_MESSAGES/django.mo
CONFLICT (content): Merge conflict in wagtail/snippets/locale/sv/LC_MESSAGES/django.mo
Auto-merging wagtail/snippets/locale/sv/LC_MESSAGES/django.po
CONFLICT (content): Merge conflict in wagtail/snippets/locale/sv/LC_MESSAGES/django.po
Auto-merging wagtail/snippets/locale/ta/LC_MESSAGES/django.po
CONFLICT (content): Merge conflict in wagtail/snippets/locale/ta/LC_MESSAGES/django.po
Auto-merging wagtail/snippets/locale/tet/LC_MESSAGES/django.mo
CONFLICT (content): Merge conflict in wagtail/snippets/locale/tet/LC_MESSAGES/django.mo

**Comment by gasman** [comment]
What's the exact `git merge` command you're trying to run? (Which branches are you trying to merge?)

Generally, there should be no need to merge across release branches when you're working with Wagtail's git repo - most of the time you'll be working on the `main` branch, and you can run `git checkout stable/6.0.x` if you want to switch to a specific release.

**Comment by arshyia3000** [comment]
Thank you Matt @gasman

I executed the following commands:

```bash
git fetch --all
git checkout stable/6.0.x
```
Up to this point, everything is fine with no conflicts, and I get the new files without conflicts. However, when I attempt to merge with the local branch, conflicts arise:

```bash
git checkout my_branch 
git merge stable/6.0.x
```

Even though I haven't changed any files, conflicts occur. 




also, I tried the following merge commands:

```bash
git merge -X theirs stable/6.0.x 
git merge -s recursive -X theirs stable/6.0.x
```


I upgraded from 5.2.2 to 5.2.3 without problem. Also, right now, I pulled origin stable/5.2.x into my local stable/5.2.x, then merge it into my_branch without problem.

**Comment by gasman** [comment, marked as answer]
If you've got a `my_branch` feature branch that was built on top of `stable/5.2.x`, then yep, merging `stable/6.0.x` isn't going to work - ever since the original 5.2 release was made and work on 6.0 started, those branches have had completely distinct histories. Generally, I'd recommend using the `main` branch as the starting point for any feature development, as that will always be where new code lands first - meaning that you won't have to switch branches to stay up to date.

To fix your current setup, I'd suggest using `git log` to find the commit hashes corresponding to the work you did on `my_branch` (as opposed to the commits that you merged in from `stable/5.2.x`). Then start a new branch based on `main`, and run `git cherry-pick [commit-hash]` for each of those commit hashes.

**Comment by arshyia3000** [comment]
Thank you Matt @gasman, I got your point
